codec-memcache/src/main/java/io/netty/handler/codec/memcache/binary/DefaultFullBinaryMemcacheRequest.java

```diff
@@ -113,6 +113,36 @@ public FullBinaryMemcacheRequest duplicate() {
         if (extras != null) {
             extras = extras.duplicate();
         }
-        return new DefaultFullBinaryMemcacheRequest(key, extras, content().duplicate());
+        return copy(key, extras, content().duplicate());
+    }
+
+    @Override
+    public FullBinaryMemcacheRequest retainedDuplicate() {
+        ByteBuf key = key();
```

Comment:
The comment was cryptic but my intention was to ask 2 questions:

`1.` Should we be doing `key.retainedDuplicate()` and `extras.retainedDuplicate()` for `retainedDuplicate()` and `replace`? I thought about for a second and what you have done makes sense.
`2.` If the answer to (1.) is yes then we can share code ... which you have done :)

In summary ... well done :)